backend/src/main/java/com/github/training/user/UserService.java

```diff
@@ -9,21 +9,49 @@
 
 import java.text.MessageFormat;
 
+/**
+ * UserService class containing business logic for {@code User.class}.
+ * This class is connecting {@code UserController.class} and {@code UserRepository.class}
+ */
 @Service
 @AllArgsConstructor
 public class UserService implements UserDetailsService {
+    /**
+     * UserRepository component containing methods to connect with MySQL database.
+     */
     UserRepository userRepository;
+    /**
+     * PasswordEncoder component responsible for encoding passwords.
+     */
     PasswordEncoder passwordEncoder;
 
+    /**
+     * Saves passed User to repository.
+     *
+     * @param user passed to be saved in repository.
+     */
     public void createUser(UserDetails user) {
         ((User) user).setPassword(passwordEncoder.encode(user.getPassword()));
         userRepository.save((User) user);
     }
 
-    public boolean userExists(String username) {
+    /**
+     * Checks if User exists by email.
+     *
+     * @param username identifier passed to identify user account.
+     * @return {@code true} if User is found or {@code false} if not.
+     */
+    public boolean userExistsByEmail(String username) {
         return userRepository.existsByEmail(username);
     }
 
+    /**
+     * Loads User, if exists in the database.
+     *
+     * @param email identifier passed to identify user account.
+     * @return {@code UserDetails} - User entity from database.
+     * @throws UsernameNotFoundException if User is not found in database.
+     */
     public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
i think a better name for this method would be 'getUserByUsername'

Reply:
Thanks, but this method is implemented from UserDetailsService from "org.springframework.security.core.userdetails" package. It is required to keep this name for class to work correctly.
![Screenshot_1391](https://github.com/MemeeMaster/training/assets/85307370/6813a4f8-0e2e-4f35-8c26-ba8a40ca0b7e)